src/auth/src/credentials/service_account.rs

```diff
@@ -32,33 +94,117 @@ use time::OffsetDateTime;
 const DEFAULT_SCOPES: &str = "https://www.googleapis.com/auth/cloud-platform";
 
 pub(crate) fn creds_from(js: serde_json::Value) -> Result<Credential> {
-    let service_account_info =
-        serde_json::from_value::<ServiceAccountInfo>(js).map_err(CredentialError::non_retryable)?;
-    let token_provider = ServiceAccountTokenProvider {
-        service_account_info,
-    };
-    let token_provider = TokenCache::new(token_provider);
-
-    Ok(Credential {
-        inner: Arc::new(ServiceAccountCredential { token_provider }),
-    })
+    let service_account_key =
+        serde_json::from_value::<ServiceAccountKey>(js).map_err(CredentialError::non_retryable)?;
+    Ok(Builder::default()
+        .service_account_key(service_account_key)
+        .build())
+}
+
+#[derive(Default)]
+pub struct Builder {
```

Comment:
The builder type needs documentation. Consider including an example too:

```suggestion
/// Creates service account credentials.
///
/// # Example
/// ```
/// let key = ServiceAccountKey {

/// };
/// let credentials = Builder::new(key).with_aud("https://pubsub.googleapis.com").build();
/// ```
#[derive(Default)]
pub struct Builder {
```